dallinger/prolific.py

```diff
@@ -281,3 +282,98 @@ def _req(self, method: str, endpoint: str, **kw) -> dict:
+                            {
+                                "title": "title",
+                                "status": "status",
+                                "created": "created-date",
+                                "expiration": "",
+                                "description": "",
+                            }
+                        ]
+                    }
+                    return self.debug_log_response(response)
+
+            elif method == "POST":
+                # method="POST", endpoint="/studies/", json=payload
+                if endpoint == "/studies/":
+                    response = {
+                        "id": "study-id",
+                        "external_study_url": "external-study-url",
+                    }
+                    return self.debug_log_response(response)
+
+                # method="POST", endpoint=f"/studies/{study_id}/transition/", json={"action": "PUBLISH"},
+                elif re.match(r"/studies/[a-z0-9]+/transition/", endpoint):
+                    return self.debug_log_response(True)
+
+            elif method == "PATCH":
+                # method="PATCH", endpoint=f"/studies/{study_id}/", json={"total_available_places": new_total},
+                response = {
+                    "items": ["https://experiment-url-1", "https://experiment-url-2"],
+                    "message": "More info about this particular recruiter's process",
+                }
+                return self.debug_log_response(response)
+
+            # method="DELETE", endpoint=f"/studies/{study_id}"
+            elif method == "DELETE":
+                response = {"status_code": 204}
+                return self.debug_log_response(response)
+
+        # method="POST", endpoint=f"/bulk-bonus-payments/{setup_response['id']}/pay/"
+        elif endpoint.startswith("/bulk-bonus-payments/"):
+            response = {"id": str(uuid4())}
+            return self.debug_log_response(response)
+
+        # method="POST", endpoint="/submissions/bonus-payments/", json=payload
+        elif endpoint.startswith("/submissions/bonus-payments"):
+            response = {"id": str(uuid4())}
+            return self.debug_log_response(response)
+
+        elif endpoint.startswith("/submissions/"):
+            # method="POST", endpoint=f"/submissions/{session_id}/transition/", json={"action": "APPROVE"},
+            if re.match(r"/submissions/[A-Za-z0-9]+/transition/", endpoint):
+                return self.debug_log_response(True)
+
+            # method="GET", endpoint=f"/submissions/{session_id}/"
+            elif re.match(r"/submissions/[A-Za-z0-9]+/", endpoint):
+                response = {
+                    "id": "id",
+                    "study_id": "study-id",
+                    "participant": "participant",
+                    "started_at": "started-at-timestamp",
+                    "status": "AWAITING REVIEW",
+                }
+                return self.debug_log_response(response)
```

Comment:
I think it would be cleaner if you added `response = None` before all the `if` statements, then removed all the `return` statements within the `if` statements, then after the `if` statements write the following:

```py
if response is None:
    raise RuntimeError("Simulated Prolific API call could not be matched.")
self.debug_log_response(response)
return response

Reply:
Absolutely :+1: Done.